appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java

```diff
@@ -278,6 +279,16 @@ void parse(MotionEvent event) {
           isDrag = true;
           drag = true;
 
+	  // Don't let MOVE extend beyond the bounds of the canvas
+	  // uf ExtendMovesOutsideCanvas is false
+	  if (
```

Comment:
Google Java style guide would not have a line break between lines 284 and 285, and betwee lines 286-288.